pkg/cluster/self_hosted.go

```diff
@@ -113,3 +114,93 @@ func (c *Cluster) newSelfHostedSeedMember() error {
 	c.logger.Infof("self-hosted cluster created with seed member (%s)", newMemberName)
 	return nil
 }
+
+func (c *Cluster) migrateBootMember() error {
+	endpoint := c.spec.SelfHosted.BootMemberClientEndpoint
+
+	c.logger.Infof("migrating boot member (%s)", endpoint)
+
+	cfg := clientv3.Config{
+		Endpoints:   []string{endpoint},
+		DialTimeout: constants.DefaultDialTimeout,
+	}
+	etcdcli, err := clientv3.New(cfg)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultRequestTimeout)
+	resp, err := etcdcli.MemberList(ctx)
+	cancel()
+	etcdcli.Close()
+	if err != nil {
+		return err
+	}
+
+	if len(resp.Members) != 1 {
+		return fmt.Errorf("boot cluster contains more than one member")
+	}
+	bootMember := resp.Members[0]
+	bootMemberID := resp.Members[0].ID
+
+	initialCluster := make([]string, 0)
+	for _, purl := range bootMember.PeerURLs {
+		initialCluster = append(initialCluster, fmt.Sprintf("%s=%s", bootMember.Name, purl))
+	}
+
+	// create the first member inside Kubernetes for migration
+	newMemberName := fmt.Sprintf("%s-%04d", c.name, c.idCounter)
+	c.idCounter++
+
+	peerURL := "http://$(MY_POD_IP):2380"
+	initialCluster = append(initialCluster, newMemberName+"="+peerURL)
```

Comment:
I will try to do it in this pr. The problem I do want to solve is that this func duplicates some code from the other two funcs in this file. I need to somehow reuse the code. but it would make this pr messier.